openedx/features/content_type_gating/block_transformers.py

```diff
@@ -61,7 +62,8 @@ def transform(self, usage_info, block_structure):
             weight_not_zero = block_structure.get_xblock_field(block_key, 'weight') != 0
             problem_eligible_for_content_gating = graded and has_score and weight_not_zero
             if problem_eligible_for_content_gating:
-                current_access = block_structure.get_xblock_field(block_key, 'group_access')
+                merged_access = block_structure.get_transformer_block_field(block_key, UserPartitionTransformer, 'merged_group_access', None)
+                current_access = merged_access.get_allowed_groups() if merged_access else block_structure.get_xblock_field(block_key, 'group_access')
```

Comment:
OK this is the real meat of the PR. Basically, ContentTypeGateTransformer was treating `group_access` as a non-inheritable field. But it is inheritable! So if the parent block has this set, we'd just look at the current block, assume it had no group_access, and set our own default.

Basically, I want to do the exact same logic as UserPartitionTransformer does in its `collect` phase. So I'm just stealing its calculated value here, and calling a method on its private helper class(!). I know this is bad. But here's a strawman fix proposal, while we discuss the right way to fix it.

Is there a good way to refactor a calculation like `merged_group_access` that two different transformers want?

Reply:
@cpennington heyo - would love your thoughts here

Reply:
I want to say that transformers can use values calculated by other transformers? As long as the ordering is correct. But the whole transformer stack thing needs some cleanup before there's a good way to standardize on those sorts of dependencies.

Reply:
OK, so then I should probably document heavily in the transformer docstring that it requires another transformer to run before it.

If the value isn't set (i.e. the user partition transformer was not run), I'd like to throw an exception then, to enforce that. Otherwise I'd need to write fallback code that would end up re-implementing large parts of that transformer. Does that seem kosher?

Reply:
Eh, maybe an exception is not worth it, because we may not find the `merged_group_access` value if there aren't partitions on the course. I'll just document that the fallback code has this bug in it, and that the transformer requires the user partition transformer. Hopefully that's enough to guide future developers.

Reply:
OK updated some docstrings and pulled this lookup code to a helper method to be able to add more clear comments around it. But same idea -- look up the user partition saved data.